bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiGenericBindingProvider.java

```diff
@@ -0,0 +1,383 @@
+import org.openhab.core.binding.BindingConfig;
+import org.openhab.core.items.Item;
+import org.openhab.core.library.items.ContactItem;
+import org.openhab.core.library.items.DimmerItem;
+import org.openhab.core.library.items.NumberItem;
+import org.openhab.core.library.items.StringItem;
+import org.openhab.core.library.items.SwitchItem;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.types.Command;
+import org.openhab.model.item.binding.AbstractGenericBindingProvider;
+import org.openhab.model.item.binding.BindingConfigParseException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * <p>
+ * This class parses the EcoData PowerDog LocalAPI item binding data. It
+ * registers as a {@link PowerDogLocalAPIGenericBindingProvider} service as
+ * well.
+ * </p>
+ * 
+ * <p>
+ * Here are some examples for valid binding configuration strings:
+ * <ul>
+ * <li><code>{ powerdoglocalapi="<serverId:arithmetic_1234567890:300000" }</code></li>
+ * <li><code>{ powerdoglocalapi="<powerdog:pv_global_1234567890:300000" }</code></li>
+ * <li><code>{ powerdoglocalapi="<powerdog:pv_global_1234567890:300000:Current_Value" }</code></li>
+ * <li><code>{ powerdoglocalapi="<powerdog:impulsecounter_1234567890:300000:Unit_1000000" }</code></li>
+ * <li><code>{ powerdoglocalapi=">powerdog:powerapi_1234567890:300000" }</code></li>
+ * </ul>
+ * 
+ * The 'serverId' referenced in the binding string is configured in the
+ * openhab.cfg file -: powerdoglocalapi:serverId.host = powerdog
+ * 
+ * 'serverId' can be any alphanumeric string as long as it is the same in the
+ * binding and configuration file. <b>NOTE</b>: The parameter is case sensitive!
+ * 
+ * @author wuellueb
+ * @since 1.8.0
+ */
+public class PowerDogLocalApiGenericBindingProvider extends
+		AbstractGenericBindingProvider implements
+		PowerDogLocalApiBindingProvider {
+
+	static final Logger logger = LoggerFactory
+			.getLogger(PowerDogLocalApiGenericBindingProvider.class);
+
+	/** {@link Pattern} which matches a binding configuration part */
+	private static final Pattern BASE_CONFIG_PATTERN = Pattern
+			.compile("(<|>)([0-9._a-zA-Z]+:[0-9._a-zA-Z:]+)");
+
+	/** {@link Pattern} which matches an In-Binding */
+	private static final Pattern INNER_BINDING_PATTERN = Pattern
+			.compile("([0-9._a-zA-Z]+):([0-9._a-zA-Z]+):([0-9]+)(:?[0-9._a-zA-Z]*)");
+
+	/**
+	 * Artificial command for the PowerDog configuration
+	 */
+	protected static final Command IN_BINDING_KEY = StringType
+			.valueOf("IN_BINDING");
+	protected static final Command OUT_BINDING_KEY = StringType
+			.valueOf("OUT_BINDING");
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public String getBindingType() {
+		return "powerdoglocalapi";
+	}
+
+	/**
+	 * @{inheritDoc
+	 */
+	@Override
+	public void validateItemType(Item item, String bindingConfig)
+			throws BindingConfigParseException {
+		if (!(item instanceof SwitchItem || item instanceof DimmerItem
```

Comment:
This one is not justified, as this is a multiline logic expression being evaluated...